travis/travis_weblate.py

```diff
@@ -1,166 +1,245 @@
+            name = '%(host)s:%(owner)s/%(repo)s' % {
+                'host': host,
+                'owner': owner,
+                'repo': repo
+            }
+        self.repo_name = name + '(' + self.branch + ')'
+        self.wl_api = WeblateApi()
+        self.gh_api = GitHubApi()
+        self._travis_home = os.environ.get("HOME", "~/")
+        self._travis_build_dir = os.environ.get("TRAVIS_BUILD_DIR", "../..")
+        self._odoo_version = os.environ.get("VERSION")
+        self._odoo_branch = os.environ.get("ODOO_BRANCH")
+        self._langs = (parse_list(os.environ.get("LANG_ALLOWED")) if
+                       os.environ.get("LANG_ALLOWED", False) else [])
+        self._odoo_full = os.environ.get("ODOO_REPO", "odoo/odoo")
+        self._server_path = get_server_path(self._odoo_full,
+                                            (self._odoo_branch or
+                                             self._odoo_version),
+                                            self._travis_home)
+        self._addons_path = get_addons_path(self._travis_home,
+                                            self._travis_build_dir,
+                                            self._server_path)
+        self._connection_context = context_mapping.get(
+            self._odoo_version, Odoo10Context)
+        self._apply_patch_odoo()
+        self._get_modules_installed()
+
+    def _check(self):
+        self.wl_api._check()
+        self.gh_api._check()
+
+    def _apply_patch_odoo(self):
+        """This patch is necessary because the weblate does not check which
+        word and the translated are the same to use it in its percentage of
+        translated"""
+        paths = [os.path.join('openerp', 'tools', 'translate.py'),
+                 os.path.join('odoo', 'tools', 'translate.py')]
+        for path in paths:
+            s_file = os.path.join(self._server_path, path)
+            if not os.path.isfile(s_file):
+                continue
+            cmd = ["sed", "-i", "-e",
+                   r"s/translation'] = src/translation'] = ''/g",
+                   s_file]
+            print " ".join(cmd)
+            subprocess.call(cmd)
+
+    def _get_modules_installed(self):
+        self._installed_modules = []
+        with self._connection_context(self._server_path, self._addons_path,
+                                      "openerp_test") as odoo_context:
+            odoo_context.cr.execute("select name from ir_module_module"
+                                    " where state = 'installed'")
+            self._installed_modules = odoo_context.cr.dictfetchall()
+
+    def _generate_odoo_po_files(self, component):
+        generated = False
+        module = []
+        with self._connection_context(self._server_path, self._addons_path,
+                                      "openerp_test") as odoo_context:
+            for install in self._installed_modules:
+                if install['name'] in component['filemask']:
```

Comment:
I made other way more exactly to check this if `install['name'] == component['name']`